core/src/autogluon/core/models/greedy_ensemble/ensemble_selection.py

```diff
@@ -110,25 +110,22 @@ def _fit(self, predictions: List[np.ndarray], labels: np.ndarray, time_limit=Non
         round_scores = False
         epsilon = 1e-4
         round_decimals = 6
+        ensemble_prediction = np.zeros(predictions[0].shape)
+        weighted_ensemble_prediction = np.zeros(predictions[0].shape)
+        fant_ensemble_prediction = np.zeros(predictions[0].shape)
         for i in range(ensemble_size):
             scores = np.zeros((len(predictions)))
             s = len(ensemble)
-            if s == 0:
-                weighted_ensemble_prediction = np.zeros(predictions[0].shape)
-            else:
-                # Memory-efficient averaging!
-                ensemble_prediction = np.zeros(ensemble[0].shape)
-                for pred in ensemble:
-                    ensemble_prediction += pred
-                ensemble_prediction /= s
-
-                weighted_ensemble_prediction = (s / float(s + 1)) * ensemble_prediction
-            fant_ensemble_prediction = np.zeros(weighted_ensemble_prediction.shape)
+
+            if s != 0:
+                ensemble_prediction *= (s - 1) / s
+                ensemble_prediction += ensemble[-1] / s
+                weighted_ensemble_prediction[:] = (s / float(s + 1)) * ensemble_prediction
```

Comment:
I moved the `ensemble_prediction` logic to before `weighted_ensemble_prediction` is updated to avoid one unused `ensemble_prediction` operation on the final iteration of the for loop.